### ISSUE TITLE
usingnamespace @cImport() ->  "error: expected type 'f64', found 'std.log.Level'"

### ISSUE THREAD

**Issue by xxxbxxx** [ziglang/zig#6513]
since 333b12a8f9beb5864fa05ce413c5b935e925ec14 ,

trying to compile a _main.zig_ interfacing with C code that uses math.h,
 using a flat namespace:
`usingnamespace @cImport({ ... })`

fails with 
```
$ ./zig build-exe -lc hello.zig 
./lib/std/log.zig:129:22: error: expected type 'f64', found 'std.log.Level'
            root.log(message_level, scope, format, args);
                     ^
./lib/std/log.zig:185:16: note: called from here
            log(.err, scope, format, args);
               ^
./lib/std/start.zig:273:28: note: called from here
                std.log.err("{}", .{@errorName(err)});
                           ^
./lib/std/start.zig:256:22: note: called from here
pub fn callMain() u8 {
                     ^
./lib/std/log.zig:80:19: note: std.log.Level declared here
pub const Level = enum {
                  ^
```

hello.zig:
```zig
usingnamespace @cImport({
    @cInclude("math.h");
});
pub fn main() !void { return error.Dummy; }
```

Just wanted to document the change (and make sure it is intended), but the fix is easy:

* either use a namespace: ```const C = @cImport(...)```  ```while (C.SDL_PollEvent(&event) != 0) {```
* or move the main.zig out of root namespace
```zig
const app = @import("hello.zig");
pub fn main() !void {
    app.main();
}
```

nb: It did happen to me because of:
```zig
usingnamespace @cImport({
    @cDefine("SDL_MAIN_HANDLED", "1");
    @cInclude("SDL2/SDL.h");
});
```

**Comment by codehz** [comment]
it seems the math.h contains a function called "log", and std.log will try to use root log function.

https://github.com/ziglang/zig/blob/e9434ff8f46ec27a863d0b071c0a99b4f01588dc/lib/std/log.zig#L126-L127

but it should not use unexported (aka without pub) function...

**Comment by IridescentRose** [comment]
I don't know necessarily the official Zig answer, but this seems like the Zig equivalent of  `using namespace std;` or bad naming in C, in that functions with common names, or names with different meanings in different context will overlap and cause issues. I think it's intentional, because normally we put things in their own namespaces, and flatly importing C into Zig sounds like a bad idea. The reason log can't be used here is that std.log can be user overridden (intended), meaning that flatly putting another function named log in the root namespace will cause issues.

@codehz Translate-C will default declarations in a .h file to things like `pub extern fn log(x: f64) f64;` since they are inherently public by being in a .h file. This is intended behavior as well. The solution is to not include the C standard library into your root namespace.

**Comment by xxxbxxx** [comment]
yeah I think the status quo is pretty much the good compromise.

but just to present the reverse side of the argument: now, ```log``` is practically a reserved keyword, even if you don't use the std log facility (or the std lib at all.)

**Comment by yamirui** [comment]
> I don't know necessarily the official Zig answer, but this seems like the Zig equivalent of `using namespace std;`
That's exactly what `usingnamespace` is. And yes, it is recommended to not import C library into your root.

Correct way to do it is to create `cmath.zig` and `usingnamespace` there, then optionally abstract C functions and import it to your root: `const cmath = @import("cmath.zig");`.

For even cleaner solution, do the above only once, then find `cimport.zig` in your `zig-cache` after compiling once, and keep a copy of it somewhere, then only select functions you care about and delete `@cInclude`. Zig does not require physical C headers to link to a C library correctly.

**Comment by codehz** [comment]
> inherently public

But according to the documentation, the pub keyword is needed, so it is a bug (or docs need to be update

**Comment by yamirui** [comment]
@codehz it is a known bug. #6445

**Comment by codehz** [comment]
> yeah I think the status quo is pretty much the good compromise.
> 
> but just to present the reverse side of the argument: now, `log` is practically a reserved keyword, even if you don't use the std log facility (or the std lib at all.)
> (well not 'keyword' but you see)

I think zig need a better way to configure these `global settings` (aka `log`, `evented_io`...)

**Comment by andrewrk** [comment]
duplicate of #6445
